### PR TITLE
WIP: Implement conversion to first-order ODEs

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/variables.jl [hunk 1]
```diff
@@ -52,6 +52,25 @@ function Base.show(io::IO, A::Variable)
     end
 end
 
+extract_idv(eq) = eq.args[1].diff.x
+function extract_symbol_order(eq)
```
Comment (ChrisRackauckas) [inline src/variables.jl]: This seems out of place since it's a function on equations instead of variables

**Comment by YingboMa** [comment]
@ChrisRackauckas Ready for code review

File: src/systems.jl [hunk 2]
```diff
@@ -4,14 +4,76 @@ struct DiffEqSystem <: AbstractSystem
     dvs::Vector{Variable}
     vs::Vector{Variable}
     ps::Vector{Variable}
+    function DiffEqSystem(eqs, idvs, dvs, vs, ps; lowered = false, kwargs...)
+        lowered_eqs = lowered ? eqs : ode_order_lowering(eqs; kwargs...)
```
Comment (ChrisRackauckas) [inline src/systems.jl]: I wouldn't try stuffing it into here. That will get messy really fast. Instead, just make it a function `new_sys = ode_order_lowering(sys)`

**Comment by ChrisRackauckas** [comment]
Instead of being built into the `DiffEqSystem` call, this should be implemented as a transformation, so it should be a function which takes in a system and spits out a system. That makes the whole transformation system easier to extend.

File: test/system_construction.jl [hunk 3]
```diff
@@ -38,71 +38,10 @@ test_vars_extraction(de, de2)
...
-# Define some variables
-@DVar x y z
-@IVar t
-@Deriv D'~t # Default of first derivative, Derivative(t,1)
-@Param σ ρ β
-@Const c=0
-@Var a
-
-# Define a differential equation
-eqs = [D*x ~ σ*(y-x),
```
Comment (ChrisRackauckas) [inline test/system_construction.jl]: where did all of this go?

File: src/systems.jl [hunk 4]
```diff
@@ -4,18 +4,12 @@ struct DiffEqSystem <: AbstractSystem
     dvs::Vector{Variable}
     vs::Vector{Variable}
     ps::Vector{Variable}
-    function DiffEqSystem(eqs, idvs, dvs, vs, ps)
```
Comment (ChrisRackauckas) [inline src/systems.jl]: isn't this constructor missing now?

**Pull request merged**